File: bogosort.py
import random
import sys


def build_deck(size: int):
  return list(range(size))

# writing our own shuffle (fisher-yates)
def fy_shuffle(array: list[int]):

  for i in reversed(array):
    j = random.randint(0, len(array) - 1)

    temp = array[i]
    array[i] = array[j]
    array[j] = temp

  return array


def bogo_sort(input: list[int]):

  sorted = False
  attempts = 0

  while not sorted:

    done = True

    tmp = fy_shuffle(input)

    for i in tmp:
      if i != tmp[i]:
        done = False

    attempts += 1

    sorted = done

  return attempts

size = int(sys.argv[1])

input = fy_shuffle(build_deck(size))
total = bogo_sort(input)
print("Attempts: ", total)

File: bogosort.js

function buildArray(size) {
  // could use spread operator but i feel like it hides too much
  // return [...Array(size).keys()]
  return Array.from(Array(size).keys());
}

function fyShuffle(list) {
  for (let i = (size - 1); i >= 0; i--) {
    let j = Math.floor(Math.random() * (size - 1));

    let tmp = list[i];
    list[i] = list[j];
    list[j] = tmp;
  }

  return list;
}

function bogosort(input) {

  sorted = true;
  attempts = 0;

  do {

    sorted = true;

    att = fyShuffle(input);

    // console.log(att);

    att.forEach(function (val, index) {
      if (val != index) {
        sorted = false;
      }
    })

    attempts += 1;

  } while (!sorted);

  // return [att, attempts];
  return attempts;

}

let size = parseInt(process.argv[2]);
let input = fyShuffle(buildArray(size));
let total = bogosort(input);
console.log("Attempts: ", total);
